ygot/render.go

```diff
@@ -415,18 +462,97 @@ func findUpdatedLeaves(leaves map[*path]interface{}, s GoStruct, parent *gnmiPat
 				errs.Add(findUpdatedLeaves(leaves, goStruct, childPath))
 			}
 		case reflect.Ptr:
-			// Determine whether this is a pointer to a struct (another YANG container), or a leaf.
-			switch fval.Elem().Kind() {
-			case reflect.Struct:
-				goStruct, ok := fval.Interface().(GoStruct)
-				if !ok {
-					errs.Add(fmt.Errorf("%v: was not a valid GoStruct", mapPaths[0]))
+			if _, ok := fval.Interface().(GoOrderedList); ok {
```

Comment:
nit: handle GoOrderedList in a separate func? it adds significant complexity to this func

Reply:
Done.